prime-router/src/main/kotlin/cli/tests/HistoryApiTest.kt

```diff
@@ -0,0 +1,233 @@
+package gov.cdc.prime.router.cli.tests
+
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.github.kittinunf.fuel.Fuel
+import com.github.kittinunf.fuel.core.extensions.authentication
+import com.github.kittinunf.result.Result
+import com.microsoft.azure.functions.HttpStatus
+import gov.cdc.prime.router.Receiver
+import gov.cdc.prime.router.ReportId
+import gov.cdc.prime.router.azure.HttpUtilities
+import gov.cdc.prime.router.cli.CommandUtilities.Companion.abort
+import gov.cdc.prime.router.cli.FileUtilities
+import gov.cdc.prime.router.cli.OktaCommand
+import gov.cdc.prime.router.common.Environment
+import java.net.HttpURLConnection
+import java.time.OffsetDateTime
+
+data class ExpectedSubmissionHistory(
+    val taskId: Int,
+    val createdAt: OffsetDateTime,
+    val sendingOrg: String,
+    val httpStatus: Int,
+    val id: ReportId?,
+    val topic: String?,
+    val reportItemCount: Int?,
+    val warningCount: Int?,
+    val errorCount: Int?
+)
+
+data class SubmissionAPITestCase(
+    val name: String,
+    val path: String,
+    val headers: Map<String, String>,
+    val parameters: List<Pair<String, Any?>>?,
+    val bearer: String,
+    val expectedHttpStatus: HttpStatus,
+    val expectedReports: Set<ReportId>,
+)
+
+class HistoryApiTest : CoolTest() {
+    override val name = "history"
+    override val description = "Test the History/Lineage API"
+    override val status = TestStatus.SMOKE
+
+    // todo this code was copied from oktaAccessToken in SettingCommands.kt
+    // This is now copied in 3 places:  here, SettingCommands, LookupTableCommands, in 3 different ways.
+    // Best style is the lazy init in SettingCommands.  To factor up we need to pass Environment to
+    // CoolTest constructor.  This is a lotta work.
+    /**
+     * The access token left by a previous login command as specified by the command line parameters
+     */
+    fun getAccessToken(environment: Environment): String {
```

Comment:
The access token needs to become global to all CLI code, so it can be used by both the test code as well as settings and lookuptable code,` however, there's a bunch of other refactoring that's involved in that.